elastic/logstash
# add http.* deprecation log

This PR updates settings.rb and java Setting class, hence, needs to merge to main and backports to 8.x

## Release notes


Add deprecation logs for http.host, http.port, http.enabled and http.environment

## What does this PR do?

- refactor deprecated alias to support obsoleted version to provide consistent deprecation warning
- add deprecation log for http.* config

## Why is it important/What is the impact to the user?

This commit gives users a clear message that http.* will be removed in next major.

## Checklist




## Author's Checklist



## How to test this PR locally

```yaml
http.port: 9600
http.enabled: false
xpack.geoip.download.endpoint: ""
```
- run logstash with the above logstash.yml
- logstash-deprecation.log should say `http.*` will be removed in version 9 and `xpack.geoip.download.endpoint` will be removed in a future release

## Related issues


- relates: https://github.com/elastic/ingest-dev/issues/3663

## Use cases



## Screenshots



## Logs
